src/AppBundle/EventSubscriber/RateLimitSubscriber.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * This file contains only the RateLimitSubscriber class.
+ */
+
+namespace AppBundle\EventSubscriber;
+
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Psr\Cache\CacheItemPoolInterface;
+use DateInterval;
+
+/**
+ * A RateLimitSubscriber checks to see if users are exceeding usage limitations.
+ */
+class RateLimitSubscriber implements EventSubscriberInterface
+{
+
+    /** @var Container The DI container. */
+    protected $container;
+
+    /**
+     * Save the container for later use.
+     * @param Container $container The DI container.
+     */
+    public function __construct(Container $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * Register our interest in the kernel.controller event.
+     * @return string[]
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            KernelEvents::CONTROLLER => 'onKernelController',
+        ];
+    }
+
+    /**
+     * Check if the current user has exceeded the configured usage limitations.
+     * @param FilterControllerEvent $event The event.
+     * @throws AccessDeniedHttpException If rate limits have been exceeded.
+     */
+    public function onKernelController(FilterControllerEvent $event)
+    {
+        $rateLimit = $this->container->getParameter('app.rate_limit_count');
+        $rateDuration = $this->container->getParameter('app.rate_limit_time');
+
+        // Zero values indicate the rate limiting feature should be disabled.
+        if ((int) $rateLimit === 0 || (int) $rateDuration === 0) {
```

Comment:
@samwilson I am typecasting into an int rather than checking with `is_int`. Does this suffice, or is it important to validate that it is actually an integer?